packages/sol-doc/src/solidity_doc_generator.ts

```diff
@@ -0,0 +1,66 @@
+import * as _ from 'lodash';
+
+import { MethodAbi } from 'ethereum-types';
+
+import { Compiler, CompilerOptions } from '@0xproject/sol-compiler';
+import { logUtils } from '@0xproject/utils';
+
+import { SolidityDocFormat } from './solidity_doc_format';
+
+export class SolidityDocGenerator {
+    private readonly _compilerOptions: CompilerOptions;
+    constructor(contractsDir: string, artifactsDir: string) {
+        // instantiate sol-compiler, passing in options to say we want abi and devdoc
+        this._compilerOptions = {
+            contractsDir,
+            artifactsDir,
+            contracts: '*',
+            compilerSettings: {
+                outputSelection: {
+                    ['*']: {
+                        ['*']: ['abi', 'devdoc'],
+                    },
+                },
+            },
+        };
+    }
+    /// run `contractsToCompile` through compiler, gathering output
+    public async generateAsync(contractsToCompile: string[]): Promise<SolidityDocFormat> {
+        if (!_.isUndefined(contractsToCompile)) {
```

Comment:
Let's assign `!_.isUndefined(contractsToCompile)` to a variable called `shouldOverrideCatchAllContractsConfig`, otherwise it's not immediately clear what happens if **no** contracts are explicitly supplied.

Reply:
I like it.